components/src/middlewares/Toastr.spec.js

```diff
@@ -0,0 +1,174 @@
+
+  beforeEach('mock cut', () => {
+    cut = new ToastrMiddleware(time);
+    state = {
+      toastr: {
+        toastNo: toastNo
+      }
+    };
+  });
+
+  describe('TO MIDDLEWARE', () => {
+    beforeEach('mock methods', () => {
+      cut.clearTimeoutFor = sinon.spy();
+      cut.toast = sinon.spy();
+    });
+
+    it('should call only clearTimeoutFor and next, when respected action is dispatched', () => {
+      // given
+      const store = null;
+      const next = sinon.spy();
+      const middleware = cut.toMiddleware()(store)(next);
+      const action = removeToast(toastNo);
+      expect(middleware).to.be.a('function');
+      expect(action).to.be.an('object');
+
+      // when
+      middleware(action);
+
+      // then
+      expect(cut.clearTimeoutFor.calledWith(toastNo)).to.be.true;
+      expect(next.calledWith(action)).to.be.true;
+      expect(cut.toast.called).to.be.false;
+    });
+
+    it('should call only next when non-respected action is dispatched', () => {
+      // given
+      const store = null;
+      const next = sinon.spy();
+      const middleware = cut.toMiddleware()(store)(next);
+      const action = { type: 'test' };
+      expect(middleware).to.be.a('function');
+      expect(action).to.be.an('object');
+
+      // when
+      middleware(action);
+
+      // then
+      expect(cut.clearTimeoutFor.called).to.be.false;
+      expect(next.calledWith(action)).to.be.true;
+      expect(cut.toast.called).to.be.false;
+    });
+
+    it('should call only next and toast, when action with meta toastr is dispatched', () => {
+      // given
+      const msg = 'test';
+      const store = null;
+      const next = sinon.spy();
+      const middleware = cut.toMiddleware()(store)(next);
+      const meta = { toastr: { msg, type: 'default' } };
+      const action = { type: 'test', payload: 'test', meta };
+      expect(middleware).to.be.a('function');
+      expect(action).to.be.an('object');
+
+      // when
+      middleware(action);
+
+      // then
+      expect(cut.clearTimeoutFor.called).to.be.false;
+      expect(next.calledWith(action)).to.be.true;
+      expect(cut.toast.calledWith(store, next, action)).to.be.true;
+    });
+  });
+
+  describe('TOAST', () => {
+    // using 'before' doesn't work
+    // it might b overwriten by the global beforeEach
+    beforeEach('spy on removeToast', () => {
+      cut.setTimeoutFor = sinon.spy();
+    });
+
+    it('should call next and setTimeoutFor', () => {
+      // given
+      const msg = 'text';
+      const type = 'default';
+      const store = { getState: () => state };
+      const next = sinon.spy();
+      const action = { meta: { toastr: {
+        msg, type
+      } } };
+
+      // when
+      cut.toast(store, next, action);
+
+      // then
+      expect(next.calledWith(addToast({
+        msg, type, toastNo
+      }))).to.be.true;
+
+      expect(cut.setTimeoutFor.calledWith(toastNo, next));
+    });
+  });
+
+  describe('SET TIMEOUT FOR', () => {
```

Comment:
Those tests are relying on too much private stuff for me. Would be better to test a general public API then to verify that some particular values are present in internal structures.
(but it's just an opinion, the tests are ok)